### PR TITLE
Add DataDictionarySource import

### LINKED ISSUE #25: Datadictionary Repository import
Add functionality to import a `DataDictionarySource` into a Repository.

- [ ]  Terminology name must be given
- [ ] Concept ID can be auto generated using terminologyName:ConceptName (as in dictionary)
- [ ] Mapping based on descriptions

### PR DESCRIPTION
**Pull request by mehmetcanay**
- Removed `object` return type due to Pylance warnings
- Added `Optional` type where None is the default value of a parameter
- Added `import_data_dictionary` function for SQL and Weaviate repositories to allow the import of `DataDictionarySource`
- Implemented the tests for the new function

closes #25

### CODE REVIEW

File: tests/test_sql_repository.py [hunk 1]
```diff
@@ -52,3 +56,11 @@ def test_get_all_sentence_embedders(self):
...
         self.assertEqual(sentence_embedders[0], "sentence-transformers/all-mpnet-base-v2")
         self.assertEqual(sentence_embedders[1], "text-embedding-ada-002")
+
+    def test_import_data_dictionary(self):
+        data_dictionary_source = DataDictionarySource(os.path.join(self.TEST_DIR_PATH, "resources", "test_data_dict.csv"), "VAR_1", "DESC")
+        self.repository.import_data_dictionary(data_dictionary_source, terminology_name="import_test")
+        terminologies = [terminology.name for terminology in self.repository.get_all_terminologies()]
+        concept_identifiers = [concept.concept_identifier for concept in self.repository.get_all_concepts()]
+        self.assertIn("import_test", terminologies)
+        self.assertIn("import_test:A", concept_identifiers)
```
Comment (tiadams) [inline tests/test_sql_repository.py:66]: Would make sense to also check if the `Mapping` descriptions are correctly stored and linked to the respective Concept

File: tests/test_weaviate_repository.py [hunk 2]
```diff
@@ -112,6 +116,15 @@ def test_terminology_and_model_specific_mappings_with_similarities(self):
...
 
+    def test_import_data_dictionary(self):
+        """Test importing a data dictionary."""
+        data_dictionary_source = DataDictionarySource(os.path.join(self.TEST_DIR_PATH, "resources", "test_data_dict.csv"), "VAR_1", "DESC")
+        self.repository.import_data_dictionary(data_dictionary_source, terminology_name="import_test")
+        terminology = self.repository.get_terminology("import_test")
+        concept = self.repository.get_concept("import_test:A")
+        self.assertEqual("import_test", terminology.name)
+        self.assertEqual("import_test:A", concept.concept_identifier)
+
```
Comment (tiadams) [inline tests/test_weaviate_repository.py:127]: Same here

File: tests/test_sql_repository.py (hunk 1 above)
Comment (mehmetcanay) [reply tests/test_sql_repository.py]: I added a loop to check the linking for each description.

File: tests/test_weaviate_repository.py (hunk 2 above)
Comment (mehmetcanay) [reply tests/test_weaviate_repository.py:141]: Also did it for this.

File: datastew/repository/sqllite.py [hunk 3]
```diff
@@ -35,6 +37,34 @@ def store_all(self, model_object_instances: List[Union[Terminology, Concept, Map
         self.session.add_all(model_object_instances)
         self.session.commit()
 
+    def import_data_dictionary(self, data_dictionary: DataDictionarySource, terminology_name: str, embedding_model: Optional[EmbeddingModel] = None):
+        terminology = Terminology(terminology_name, terminology_name)
```
Comment (tiadams) [inline datastew/repository/sqllite.py:41]: I would move the storage logic down. If for example there is any issue with the data dictionary and the function fails at that point, we will have an empty terminology stored here without any links to actual concepts.

If the user then tries to redo the process with a fixed data dictionary, their initial preffered terminology name will already be taken and not available.

File: datastew/repository/weaviate.py [hunk 4]
```diff
@@ -62,7 +64,35 @@ def _create_schema_if_not_exists(self, schema):
         except Exception as e:
             raise RuntimeError(f"Failed to check/create schema for {class_name}: {e}")
 
-    def store_all(self, model_object_instances):
+    def import_data_dictionary(self, data_dictionary: DataDictionarySource, terminology_name: str, embedding_model: Optional[EmbeddingModel] = None):
+        terminology = Terminology(terminology_name, terminology_name)
+        self.store(terminology)
+        data_frame = data_dictionary.to_dataframe()
```
Comment (tiadams) [inline datastew/repository/weaviate.py:70]: Same here - would also make sense to handle this defensively with a try/catch block

File: tests/test_sql_repository.py [hunk 5]
```diff
@@ -52,3 +56,20 @@ def test_get_all_sentence_embedders(self):
         self.assertEqual(len(sentence_embedders), 2)
         self.assertEqual(sentence_embedders[0], "sentence-transformers/all-mpnet-base-v2")
         self.assertEqual(sentence_embedders[1], "text-embedding-ada-002")
+
+    def test_import_data_dictionary(self):
+        data_dictionary_source = DataDictionarySource(os.path.join(self.TEST_DIR_PATH, "resources", "test_data_dict.csv"), "VAR_1", "DESC")
```
Comment (tiadams) [inline tests/test_sql_repository.py:61]: What happens in edge cases - e.g. an empty variable or description field

Comment (mehmetcanay) [reply tests/test_sql_repository.py:61]: I'm utilizing the function from `parsing.py` that by default removes any row if one of the entries is missing.

File: datastew/repository/sqllite.py (hunk 3 above)
Comment (mehmetcanay) [reply datastew/repository/sqllite.py]: Definitely makes sense, I did not see it. Thanks for pointing out.
